src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -33,20 +34,50 @@ public PythonCodeMigrationEventArgs(string oldCode, string newCode)
 
     public abstract class PythonNodeBase : VariableInputNode
     {
-        private PythonEngineVersion engine = PythonEngineVersion.Unspecified;
+        private string engine = string.Empty;
 
         [JsonConverter(typeof(StringEnumConverter))]
         [JsonProperty(DefaultValueHandling = DefaultValueHandling.Populate)]
-        [DefaultValue(nameof(PythonEngineVersion.IronPython2))]
+        // Set the default EngineName value to IronPython2 so that older graphs can show the migration warnings.
+        [DefaultValue("IronPython2")]
+
+        // When removing this property also replace the serialized property in EngineName
+        // (i.e remove XmlIgnore and add [JsonProperty("Engine", DefaultValueHandling = DefaultValueHandling.Populate)]
         /// <summary>
         /// Return the user selected python engine enum.
         /// </summary>
+        [Obsolete("This property will be deprecated in Dynamo 3.0. Please use EngineName instead")]
         public PythonEngineVersion Engine
+        {
+            get
+            {
+                if (!Enum.TryParse(engine, out PythonEngineVersion engineVersion) ||
+                    engineVersion == PythonEngineVersion.Unspecified)
+                {
+                    // This is a first-time case for newly created nodes only
+                    SetEngineByDefault();
+                }
+                return engineVersion;
+            }
+            set
+            {
+                engine = value.ToString();
+                RaisePropertyChanged(nameof(EngineName));
+            }
+        }
+
+        [XmlIgnore]
```

Comment:
this is needed because we serialize to preferences or...?

Reply:
The reason is so that we do not searialize 2 properties that represent the same model data.
When EngineName will be removed we can only use Engine for serialization.
See explanation from code at line 44:
```
// When removing this property also replace the serialized property in EngineName
// (i.e remove XmlIgnore and add [JsonProperty("Engine", DefaultValueHandling = DefaultValueHandling.Populate)]
        ```